src/stateful-core.ts

```diff
@@ -24,7 +24,23 @@ import { Unsubscribe } from '@reduxjs/toolkit';
 import { isLevenshteinMatch } from './utils/searchable-facets';
 
 export default class StatefulCore {
-  constructor(private core: AnswersCore, private stateManager: StateManager) {}
+  private latestRequestIds: Record<string, number>;
+
+  constructor(private core: AnswersCore, private stateManager: StateManager) {
+    this.latestRequestIds = {};
+  }
+
+  private getNewRequestId(requestName: string): number {
```

Comment:
maybe `updateRequestId`? also okay with the current name though! think I'm splitting hairs here

Reply:
yeah it does update request id internally, so your suggestion make sense. Updated!